### ISSUE TITLE
Need to run linter TWICE for fixing to work

### ISSUE THREAD

**Issue by rich-newman** [palantir/tslint#2835]
### Bug Report

- __TSLint version__: 5.3.2
- __TypeScript version__: 2.3.2
- __Running TSLint via__: CLI

__New bug: this behavior was not present in TSLint version 5.2.0__

#### TypeScript code being linted

```ts
function foo() {
    var i = "test";
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:recommended"]
}
```

Two rules that affect this are in place: prefer-const and no-var-keyword.

#### Actual behavior

Run tslint with --fix.  First pass it updates the code to:

```ts
function foo() {
    let i = "test";
}
```

It still reports an error for prefer-const, which has a fixer.
Run tslint again with --fix.  Code is updated to:

```ts
function foo() {
    const i = "test";
}
```

#### Expected behavior

Should update to final version the first time we lint with --fix.

**Comment by ajafff** [comment]
this is a duplicate of #2112

In theory fixes could be applied until there are no more fixes available. That might take a long time to complete. If you have rules that apply conflicting fixes, it will never finish.
This needs a good proposal before the current behavior will be changed.

**Comment by rich-newman** [comment]
Ah, thanks, I see the problem now.  In 5.2 it looks like we were applying rules in the order in tslint.json, which works fine in this case, although accidentally because 'no-var-keyword' is before 'prefer-const' in the alphabet.  In 5.3 we are applying some other ordering, which in this case is the wrong one.  I think.

Maybe allowing rule order to be specified is a reasonable approach, at least initially?  There won't be too many cases where that won't be enough, in fact it's hard to think of any with the current simple rules.

I think #2112 is different by the way. prefer-const doesn't seem to work at all if the statement is outside of a function, which is what the OP is complaining about.

**Comment by rich-newman** [comment]
I realized that what's above is wrong: this is just a bug in the prefer-const fixer.  If it did what it's supposed to rule order or the number of times they are applied wouldn't matter.  

If you declare a variable with 'let' in a function the rule gives the error "use 'const' instead of 'let'" and the fixer fixes it.  If you declare a variable with 'var' in a function the rule gives the error "use 'const' instead of 'var'" but then the fixer doesn't do anything.  

It appeared to work with no-var-keyword only because the other rule was running first and converting all 'var' to 'let'. 

I'll close this issue and raise another that describes the problem without all the confusion.

**Comment by rich-newman** [comment]
Corrected issue is #2843
